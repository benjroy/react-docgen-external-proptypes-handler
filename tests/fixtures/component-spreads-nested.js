import React from 'react';
import PropTypes from 'prop-types';
import { propTypes as propTypesApi3, defaultProps as propTypesApi3 } from './api-3';

const ComponentSpreadsNested = props => (<div>component-spreads-nested</div>);

ComponentSpreadsNested.propTypes = {
  ...propTypes,
  'local-number': PropTypes.number,
};
ComponentSpreadsNested.defaultProps = {
  ...defaultProps,
  'local-number': 4242,
};

export default ComponentSpreadsNested;
