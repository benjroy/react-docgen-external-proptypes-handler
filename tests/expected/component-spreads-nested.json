{
  "description": "",
  "displayName": "ComponentSpreadsNested",
  "methods": [],
  "props": {
    "api-1-node-required": {
      "type": {
        "name": "node"
      },
      "required": true,
      "description": "Comment for api-1-node-required."
    },
    "api-1-bool": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "Comment for api-1-bool.",
      "defaultValue": {
        "computed": false,
        "value": false
      }
    },
    "api-2-node-required": {
      "type": {
        "name": "node"
      },
      "required": true,
      "description": "Comment for api-2-node-required."
    },
    "api-2-string": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Comment for api-2-string.",
      "defaultValue": {
        "computed": false,
        "value": "api-2-string-value"
      }
    },
    "api-2-node-required": {
      "type": {
        "name": "node"
      },
      "required": true,
      "description": "Comment for api-2-node-required."
    },
    "local-number": {
      "type": {
        "name": "number"
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "4242",
        "computed": false
      }
    }
  },
  "composes": ["./api-1", "./api-2"]
}
